COORDINATE_SYSTEM2/COORDINATE_SYSTEM2/CoordinateSystem.cs

```diff
@@ -6,48 +6,123 @@
 
 namespace COORDINATE_SYSTEM2
 {
+    /// <summary>
+    /// Клас Координатна система.
+    /// Містить Ліст всіх створених точок та Ліст всіх знайдених дистанцій.
+    /// Містить Add(), який створює нову точку, добавляє її в Ліст, шукає відстані 
+    /// між існуючими точками до нової точки і добавляє їх в ліст.
+    /// Містить DisplayAllPoints(), який друкує всі точки з ліста.
+    /// Містить MaxDistance() для пошуку максимальної відстані між точками.
+    /// Містить PrintAllDistances(), який друкує всі наявні відстані з Ліста.
+    /// </summary>
     internal class CoordinateSystem
     {
-        List<Point> listOfPoints = new List<Point>();
+        List<IFigure> listOfFigures = new List<IFigure>();
 
-        List<PointsDistance> pointsDistances = new List<PointsDistance>();
+        List<PointsDistance> listOfPointsDistances = new List<PointsDistance>();
 
-        public void Add(Point point)
+        public void Add(IFigure figure)
         {
-            foreach (Point point1 in listOfPoints)
+            listOfFigures.Add(figure);
+        }
+
+        public void PrintAreas()
+        {
+            foreach (IFigure figure in listOfFigures)
             {
-                PointsDistance distance = new PointsDistance(point, point1);
-                pointsDistances.Add(distance);
+                var area = figure.Area();
+                var pishovNahyi = area == 0 ? "N / A" : area.ToString();
+                Console.WriteLine(figure.ToString());
+                Console.WriteLine($"Area is {pishovNahyi}\n");
             }
+        }
 
-            listOfPoints.Add(point);
+        public void PrintPerimeter()
+        {
+            foreach (IFigure figure in listOfFigures)
+            {
+                var perimeter = figure.Perimeter();
+                var pishovNahyi = perimeter == 0 ? "N / A" : perimeter.ToString();
+                Console.WriteLine(figure.ToString());
+                Console.WriteLine($"Perimeter is {pishovNahyi}\n");
+            }
         }
 
         public void DisplayAllPoints()
         {
+            List<Point> listOfPoints = AddAllFiguresPointsIntoList();
+
             for (int i = 0; i < listOfPoints.Count; i++)
             {
                 Console.WriteLine($"point {i} : {listOfPoints[i]};");
             }
         }
+        
+        public List<Point> AddAllFiguresPointsIntoList()
```

Comment:
no need to keep this method public, it's for internal use only so make it private pls.

Reply:
Also about the method name `AddAllFiguresPointsIntoList`. If I'm calling a method called Add I expect parameters that I want to add like `void Add(IFigure figure)`. Change the name. smth like `GetFiguresPoints`